### PR TITLE
App 13009 icons page design improvements

### PR DESCRIPTION
**Pull request by webspaceadam**
## Summary

- [Ticket](https://flipapp.myjetbrains.com/youtrack/issue/APP-13009/Icons-Page-Design-Improvements)

### CODE REVIEW

File: apps/swirl-docs/src/components/Icons/IconDownload.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+  const { getReferenceProps, getFloatingProps } = useInteractions([
+    click,
+    dismiss,
+    role,
+  ]);
+
+  function handleActionItemClick(size: number): void {
+    setIconPixelSize(size);
+    setIsOpen(false);
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Icons/IconDownload.tsx:53]: **thought:** Should we start the download here immediately?

File: apps/swirl-docs/src/components/Icons/IconDownload.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          "bg-surface-info-default hover:bg-surface-highlight-hovered active:bg-surface-highlight-pressed",
+          {
+            "rounded-border-radius-sm": !isOpen,
+            "rounded-t-border-radius-sm": isOpen,
+          }
+        )}
+      >
+        <button
+          className="w-full max-h-6 text-center"
+          onClick={() => handleIconDownload()}
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Icons/IconDownload.tsx:78]: **suggestion:** `type="button"` to be safe and not trigger a form submit.

File: apps/swirl-docs/src/components/Icons/IconDownload.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        >
+          {isLoadingIconDownload ? (
+            <SwirlSpinner label="loading icon from server" size="s" />
+          ) : (
+            `Download ${iconPixelSize}px Icon`
+          )}
+        </button>
+        <button
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Icons/IconDownload.tsx:86]: **suggestion:** `type="button"`

File: apps/swirl-docs/src/components/Icons/IconDownload.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+          ref={reference}
+          id="icon-size-trigger"
+          className={classNames(
+            "relative w-6 h-6 mx-2",
+            "before:absolute before:top-0 before:left-[-8px] before:h-[24px] before:w-[1px] before:bg-text-on-action-primary"
+          )}
+          onClick={() => console.log("clicked")}
+          {...getReferenceProps()}
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Icons/IconDownload.tsx:94]: **issue:** Missing label.

File: apps/swirl-docs/src/components/Icons/IconDownload.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+      </div>
+      {isOpen && (
+        <FloatingFocusManager context={context} modal={false}>
+          <div
+            className="Popover w-full border border-border-default rounded-b-border-radius-sm"
+            ref={floating}
+            style={{
+              position: strategy,
+            }}
+            aria-label="Icon Sizes"
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Icons/IconDownload.tsx:107]: **issue:** `aria-label` does not work on a div. Should be removed.

File: apps/swirl-docs/src/components/Icons/IconGridItem.tsx [hunk 6]
```diff
@@ -37,14 +40,21 @@ const IconGridItem: FunctionComponent<IconGridProps> = ({
           aria-label={`${icon}-icon`}
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Icons/IconGridItem.tsx:40]: **suggestion:** I think the button text should be enough here.

File: apps/swirl-docs/src/components/Icons/IconInfo.tsx [hunk 7]
```diff
@@ -1,43 +1,34 @@
...
-            }}
-          ></code>
-        </pre>
-      </div>
+      <IconDownload icon={icon} />
+      <hr className="border-b-1 my-6" />
+      <h2 className="font-semibold text-base mb-2">
+        <a
+          className="inline-flex font-semibold text-font-size-base text-text-default"
+          href="https://swirl-storybook.flip-app.dev/?path=/docs/components-flipicon--flip-icon"
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Icons/IconInfo.tsx:24]: **issue:** `?path=/docs/components-swirlicon--swirl-icon`

**Review by Sqrrl — CHANGES_REQUESTED** [review]
- Icon sidebar is missing on desktop viewport
- The mobile popover backdrop doesn't overlay the entire page (see screenshot). Maybe you could use a portal to move it out of its container.
- The component links need to be renamed (flip -> swirl).
- Firefox has issues displaying the icon grid correctly (see second screenshot).

Bildschirm­foto 2023-01-16 um 09 55 06

Bildschirm­foto 2023-01-16 um 10 11 28

File: apps/swirl-docs/src/components/Icons/IconDownload.tsx (hunk 1 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/components/Icons/IconDownload.tsx:53]: good idea. implemented

**Comment by webspaceadam** [comment]
Yeah older firefox does not support container queries.... need to find a workaround for that. Thought 110 was already released, but nope. just in RC-Phase

**Review by Sqrrl — APPROVED** [review]

**Pull request merged**